src/acl_usm.cpp

```diff
@@ -138,6 +144,13 @@ CL_API_ENTRY void *CL_API_CALL clHostMemAllocINTEL(
   }
 
   if (acl_get_hal()->host_alloc) {
+    std::vector<mem_properties_t> mmd_properties;
+    if (mem_id) {
+      mmd_properties.push_back(AOCL_MMD_MEM_PROPERTIES_BUFFER_LOCATION);
+      mmd_properties.push_back(mem_id.value());
```

Comment:
`std::optional` overloads [operator*](https://en.cppreference.com/w/cpp/utility/optional/operator*), which is preferable since it does not throw an exception.
```suggestion
      mmd_properties.push_back(*mem_id);
```
So the common optional pattern is to check whether a value is present, then safely access it using `operator*`.